katsdpingest/sigproc.py

```diff
@@ -1442,12 +1490,12 @@ def __init__(
                     template.percentiles_flags.instantiate(
                         command_queue, (kept_channels, baselines), prange))
             else:
-                self.percentiles.append(
-                    template.nan_percentiles.instantiate(command_queue, (5, kept_channels)))
-                self.percentiles[-1].set_value(np.nan)
-                self.percentiles_flags.append(
-                    template.zero_percentiles_flags.instantiate(command_queue, (kept_channels,)))
-                self.percentiles_flags[-1].set_value(0)
+                p = template.nan_percentiles.instantiate(command_queue, (5, kept_channels))
```

Comment:
This bit seems like more than just type annotation...

Reply:
If I did it right then it's just changing from "append this thing, then modify it" to "create this thing, modify it, and append it" i.e. there should be no difference. I needed to make that split because the list is typed with a base class, so if I do `self.percentiles_flags[-1]` then mypy only knows that it's an `Operation`, not a `Fill`.